src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -7975,12 +7978,7 @@ private BoundExpression BindIndexedPropertyAccess(SyntaxNode syntax, BoundExpres
                             property.OriginalDefinition is { ParameterCount: 1 } original &&
                             original.Parameters[0] is { Type: { SpecialType: SpecialType.System_Int32 }, RefKind: RefKind.None })
                         {
-<<<<<<< HEAD
-=======
-                            CheckImplicitThisCopyInReadOnlyMember(receiverOpt, lengthOrCountProperty.GetMethod, diagnostics);
```

Comment:
This call to `CheckImplicitThisCopyInReadOnlyMember` turned out unnecessary because the caller (for indexing via `this[int]`) already reported exactly the same diagnostic (see line 7932 above in diff).
Note that the other caller is for list-patterns (`TryFindIndexerOrIndexerPattern` in `Binder_Patterns.cs`). Added a test for that scenario (`PatternIndexRangeReadOnly_01`) but I don't think it should report this diagnostic anyway because the indexer access is off of a special receiver for patterns... `CheckImplicitThisCopyInReadOnlyMember` only reports when the receiver is `this`.